data/data/agent/files/usr/local/bin/release-image.sh.template

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env bash
+# This library provides an `image_for` helper function which can get the
+# pull spec for a specific image in a release.
+
+image_for() {
+   podman run --rm --net=none "${RELEASE_IMAGE_DIGEST}" image "${1}"
+}
+
+
+ImageType=${2}
+ImagesEnvFile=${1}/agent-images.env
+
+# Check if we have already found this image
+case ${ImageType} in
+     agent-installer-api-server)
+          if grep -q "SERVICE_IMAGE" ${ImagesEnvFile}; then
+             return
+          fi
+          ;;
+     agent-installer-csr-approver)
+          if grep -q "CONTROLLER_IMAGE" ${ImagesEnvFile}; then
+             return
+          fi
+          ;;
+     agent-installer-orchestrator)
+          if grep -q "INSTALLER_IMAGE" ${ImagesEnvFile}; then
+             return
+          fi
+          ;;
+     agent-installer-node-agent)
+          if grep -q "AGENT_IMAGE" ${ImagesEnvFile}; then
```

Comment:
This is designed to guard against two different systemd services running this script simultaneously, but this is not exactly re-entrant.
It would be better if the script just got all of the images, wrote them to a temporary file (or files) and then did a `mv` to put it in place.

Reply:
Fixed to just have one script get all the images and that script is only called when in the pod service.